mutiny-core/src/lib.rs

```diff
@@ -235,6 +235,10 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                         }
                     }
                 }
+
```

Comment:
shouldn't the `read_fut` above handle `Err` so it breaks out of the loop and disconnects?

Reply:
Don't see what you're talking about, looks like every error is handled, the disconnect is outside the inner `loop`

Reply:
when read errors happen it just keeps going inside the inner loop.

Reply:
Ah I see, good call out